operator/pkg/metrics/server.go

```diff
@@ -0,0 +1,211 @@
+			<-time.After(2 * time.Second)
+			continue
+		}
+		break
+	}
+
+	klog.Infof("Keptn Metrics server enabled: %v", serverEnabled)
+
+	if serverEnabled && m.server == nil {
+		klog.Infof("serving metrics at localhost:9999/metrics")
+
+		router := mux.NewRouter()
+		router.Path("/metrics").Handler(promhttp.Handler())
+		router.Path("/api/v1/metrics/{namespace}/{metric}").HandlerFunc(returnMetric)
+
+		m.server = &http.Server{
+			Addr:    ":9999",
+			Handler: router,
+		}
+
+		m.recordMetrics()
+
+		go func() {
+			err := m.server.ListenAndServe()
+			if err != nil {
+				klog.Errorf("could not start keptn-metrics server: %w", err)
+			}
+		}()
+
+	} else if !serverEnabled && m.server != nil {
+		if err := m.ShutDownServer(); err != nil {
+			return fmt.Errorf("could not shut down keptn-metrics server: %w", err)
+		}
+	}
+	return nil
+}
+
+func returnMetric(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+	metric := vars["metric"]
+
+	scheme := runtime.NewScheme()
+	if err := metricsv1alpha1.AddToScheme(scheme); err != nil {
+		fmt.Println("failed to add keptn-metrics to scheme: " + err.Error())
+	}
+	cl, err := ctrlclient.New(config.GetConfigOrDie(), ctrlclient.Options{Scheme: scheme})
+	if err != nil {
+		fmt.Println("failed to create client")
+		os.Exit(1)
+	}
+	metricObj := metricsv1alpha1.KeptnMetric{}
+	err = cl.Get(context.Background(), types.NamespacedName{Name: metric, Namespace: namespace}, &metricObj)
+	if err != nil {
+		fmt.Println("failed to list keptn-metrics" + err.Error())
+	}
+
+	data := map[string]string{
+		"namespace": namespace,
+		"metric":    metric,
+		"value":     metricObj.Status.Value,
+	}
+
+	err = json.NewEncoder(w).Encode(data)
+	if err != nil {
+		fmt.Println("failed to encode data")
+		os.Exit(1)
+	}
+}
+
+func (m *serverManager) recordMetrics() {
+	go func() {
+		scheme := runtime.NewScheme()
+		if err := metricsv1alpha1.AddToScheme(scheme); err != nil {
+			fmt.Println("failed to add keptn-metrics to scheme: " + err.Error())
+		}
+
+		cl, err := ctrlclient.New(config.GetConfigOrDie(), ctrlclient.Options{Scheme: scheme})
+		if err != nil {
+			fmt.Println("failed to create client")
+			os.Exit(1)
+		}
+
+		for {
+			if m.server == nil {
+				return
+			}
+			list := metricsv1alpha1.KeptnMetricList{}
+			err := cl.List(context.Background(), &list)
+			if err != nil {
+				fmt.Println("failed to list keptn-metrics" + err.Error())
+			}
+			for _, metric := range list.Items {
+				normName := CleanUpString(metric.Name)
+				if _, ok := metrics.gauges[normName]; !ok {
+					metrics.gauges[normName] = prometheus.NewGauge(prometheus.GaugeOpts{
+						Name: normName,
+						Help: metric.Name,
+					})
+					prometheus.MustRegister(metrics.gauges[normName])
+				}
+				val, _ := strconv.ParseFloat(metric.Status.Value, 64)
+				metrics.gauges[normName].Set(val)
+			}
+			<-time.After(10 * time.Second)
+		}
+	}()
+}
+
+func CleanUpString(s string) string {
```

Comment:
please add a short description about the purpose of this function. Also, let's add a unit test for this little guy ;)